Description: Applications were not able to load the PPD file from the CUPS
 queues for IPP network printers as they were confused with remote CUPS queues
Author: Michael Sweet <msweet@apple.com>
Bug-Upstream: https://www.cups.org/str.php?L4725
Last-Update: 2015-10-08
--- a/cups/util.c
+++ b/cups/util.c
@@ -1528,10 +1528,9 @@
     }
 
     if (device_uri &&
-        (!strncmp(device_uri, "ipp://", 6) ||
-         !strncmp(device_uri, "ipps://", 7) ||
-         ((strstr(device_uri, "._ipp.") != NULL ||
-           strstr(device_uri, "._ipps.") != NULL) &&
+        (((!strncmp(device_uri, "ipp://", 6) || !strncmp(device_uri, "ipps://", 7)) &&
+	  (strstr(device_uri, "/printers/") != NULL || strstr(device_uri, "/classes/") != NULL)) ||
+         ((strstr(device_uri, "._ipp.") != NULL || strstr(device_uri, "._ipps.") != NULL) &&
           !strcmp(device_uri + strlen(device_uri) - 5, "/cups"))))
     {
      /*
